iceberg/iceberg-common/src/main/java/org/apache/gravitino/iceberg/common/ops/IcebergCatalogWrapper.java

```diff
@@ -303,18 +348,38 @@ private void closePostgreSQLCatalogResource() {
   }
 
   // Some io and security configuration should pass to Iceberg REST client
-  private LoadTableResponse injectTableConfig(Supplier<LoadTableResponse> supplier) {
+  private LoadTableResponse injectTableConfig(
+      Supplier<LoadTableResponse> supplier, Function<String, Map<String, String>> configInjector) {
     LoadTableResponse loadTableResponse = supplier.get();
     return LoadTableResponse.builder()
         .withTableMetadata(loadTableResponse.tableMetadata())
-        .addAllConfig(getCatalogConfigToClient())
+        .addAllConfig(configInjector.apply(loadTableResponse.tableMetadata().location()))
         .build();
   }
 
-  private Map<String, String> getCatalogConfigToClient() {
+  private Map<String, String> getCatalogConfigToClient(String location) {
     return catalogConfigToClients;
   }
 
+  private Map<String, String> vendCredentials(String location) {
+    // ifPresentOrElse is not supported in Java8
+    if (credentialProvider.isPresent()) {
```

Comment:
You can change to simplify your code here:

```java

if (!xx.isPresent) {
  throw new XXXException();
}

....
}

Reply:
done